tests/integration/targets/lookup_collection_version/runme.yml

```
@@ -0,0 +1,34 @@
+- hosts: localhost
+  tasks:
+    - name: Test collection_version
+      assert:
+        that:
+          # Collection that does not exist
+          - query('community.general.collection_version', 'foo.bar') == [none]
+          - lookup('community.general.collection_version', 'foo.bar', result_not_found='foo') == 'foo'
+          # Collection that exists
+          - lookup('community.general.collection_version', 'community.general') is string
+          # Local collection
+          - lookup('community.general.collection_version', 'testns.testcoll') == '0.0.1'
+          # Local collection with no version
+          - lookup('community.general.collection_version', 'testns.testcoll_nv') == '*'
+          - lookup('community.general.collection_version', 'testns.testcoll_nv', result_no_version='') == ''
+          # Local collection with MANIFEST.json
+          - lookup('community.general.collection_version', 'testns.testcoll_mf') == '0.0.1'
+          # Local collection with no galaxy.yml and no MANIFEST.json
+          - lookup('community.general.collection_version', 'testns.testcoll_nothing') == '*'
+          - lookup('community.general.collection_version', 'testns.testcoll_nothing', result_no_version='0.0.0') == '0.0.0'
+          # Multiple collection names at once
+          - lookup('community.general.collection_version', 'testns.testcoll', 'testns.testcoll_nv', 'testns.testcoll_nv', 'testns.testcoll_mf', 'foo.bar')
+            == ['0.0.1', '*', '*', '0.0.1', none]
+
+    - name: Invalid FQCN
+      set_fact:
+        test: "{{ query('community.general.collection_version', 'foo.bar.baz') }}"
+      ignore_errors: true
+      register: invalid_fqcn
+
+    - name: Validate error message
+      assert:
+        that:
+          - invalid_fqcn.msg == '"foo.bar.baz" is not a FQCN'
```

```suggestion
          - "'\"foo.bar.baz\" is not a FQCN' in invalid_fqcn.msg
```
or maybe
```suggestion
          - invalid_fqcn.msg is search('"foo\.bar\.baz" is not a FQCN')
```
(I think the tests are failing because of this)

---

Yep, that solves it. (I used the `in` variant, but without escaping :) )

---

in the commit you mention fixing for core < 2.12; what change happened in 2.12 that allows `==` to be a partial match?

---

I think the behavior changed in this commit: https://github.com/ansible/ansible/commit/84e473a26ea95afec207409c42ba872c009fe6b6#diff-8900ff472220a4c25fcc888ae62fdda2b036e5f20b343fb58819f97c6b82cc22R1011 Previously lookup errors were - as any other error - wrapped in some message (https://github.com/ansible/ansible/commit/84e473a26ea95afec207409c42ba872c009fe6b6#diff-8900ff472220a4c25fcc888ae62fdda2b036e5f20b343fb58819f97c6b82cc22R1025).